src/components/InMoreLanguageSwitch.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+	<div class="wikibase-termbox-subsection-switch">
+		<div class="wikibase-termbox-subsection-switch__switch">
+			<a :href="link" @click="clientShowAllLanguages( $event )">{{ message( MESSAGE_KEYS.ALL_LANG ) }}</a>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { VueConstructor } from 'vue';
+import Component, { mixins } from 'vue-class-component';
+import Messages, { MessagesMixin } from './mixins/Messages';
+
+@Component
+export default class TermBox extends ( mixins( Messages ) as VueConstructor<MessagesMixin> ) {
+	public readonly link: string;
+	constructor() {
+		super();
+		this.link = 'place-holder-link';
+	}
+
+	public clientShowAllLanguages( event: MouseEvent ) {
+		console.log( 'TODO flip all-languages.' );
+		if ( event ) {
```

Comment:
I would have hoped that using a solid framework rids us from such conditionals. Did you encounter a situation where there was no event?

Reply:
this was just what was the the doc says...(more or less) see: https://vuejs.org/v2/guide/events.html#Event-Modifiers (last code block)....and it is changed